From c7bfc6982ace60b832e2c6bc3aaf15b496509499 Mon Sep 17 00:00:00 2001
From: Haihua Hu <jared.hu@nxp.com>
Date: Tue, 2 May 2017 10:50:35 +0800
Subject: [PATCH 2/2] glupload: passthrough composition caps features in
 physmem upload

also passthrough composition caps features in directviv upload
when from src to sink

Upstream-Status: Backport [1.12.0]
https://bugzilla.gnome.org/show_bug.cgi?id=782046

Signed-off-by: Haihua Hu <jared.hu@nxp.com>
---
 gst-libs/gst/gl/gstglupload.c | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/gst-libs/gst/gl/gstglupload.c b/gst-libs/gst/gl/gstglupload.c
index 3ef3b19..1f34a14 100644
--- a/gst-libs/gst/gl/gstglupload.c
+++ b/gst-libs/gst/gl/gstglupload.c
@@ -1199,8 +1199,13 @@ _physical_buffer_upload_transform_caps(gpointer impl, GstGLContext *context,
     gst_caps_unref (ret);
     ret = tmp;
   } else {
-    ret = gst_caps_from_string (GST_VIDEO_CAPS_MAKE_WITH_FEATURES
+    GstCaps *tmp;
+    tmp = gst_caps_from_string (GST_VIDEO_CAPS_MAKE_WITH_FEATURES
             (GST_CAPS_FEATURE_MEMORY_SYSTEM_MEMORY,GST_GL_DIRECTVIV_FORMAT));
+    ret =
+        _set_caps_features_with_passthrough (tmp,
+        GST_CAPS_FEATURE_MEMORY_SYSTEM_MEMORY, passthrough);
+    gst_caps_unref (tmp);
   }
 
   gst_caps_features_free (passthrough);
-- 
1.9.1

